File: pset6/readability/readability.py
from cs50 import get_string

# Get input from user
text = get_string("Text: ")

allcount = len(text)

# Letters counting
count_letters = 0

alphabet = [ "a", "b", "c", "d", "e", "f", "g", "h",
             "i", "j", "k", "l", "m", "n", "o", "p",
             "q", "r", "s", "t", "u", "v", "w", "x",
             "y", "z"]

for i in range(allcount):
    if text[i].lower() in alphabet:
        count_letters += 1

# Words counting
count_words = 0

for i in range(allcount):
    if text[i] == " " and not text[i + 1] == " ":
        count_words += 1

count_words += 1

# Sentences counting
count_sentences = 0

for i in range(allcount):
    if text[i] == "." or text[i] == "!" or text[i] == "?":
        count_sentences += 1

# Coleman-Liau index
x = count_letters
y = count_words
z = count_sentences

L = float(100 * (x / y))
S = float(100 * (z / y))

index = (0.0588 * L) - (0.296 * S) - 15.8
grade = int(round(index))

# Grade classification
if grade < 1:
    print("Before Grade 1")
elif grade >= 16:
    print("Grade 16+")
else:
    print("Grade %d" % grade)

File: pset2/readability/readability.c
#include <stdio.h>
#include <cs50.h>
#include <ctype.h>
#include <math.h>
#include <string.h>

int main(void)
{
//Get the text input from user
    string text = get_string("Text: ");

//Letters counting
    int count_letters = 0;
    int count_blank = 0;
    int count_digit = 0;
    int count_punct = 0;
    int tmp_letters = 0;

    tmp_letters = strlen(text);

    for (int a = 0; text[a]; a++)
    {
        if (isblank(text[a]))
        {
            count_blank++;
        }
    }
    for (int b = 0; text[b]; b++)
    {
        if (isdigit(text[b]))
        {
            count_digit++;
        }
    }
    for (int c = 0; text[c]; c++)
    {
        if (ispunct(text[c]))
        {
            count_punct++;
        }
    }
    count_letters = tmp_letters - count_blank - count_digit - count_punct;

//Words counting
    int count_words = 0;

    for (int d = 0; text[d] != '\0'; d++)
    {
        if (text[d] == ' ' && text[d + 1] != ' ')
        {
            count_words++;
        }
    }
    count_words += 1;

//Sentences counting
    int count_sentences = 0;

    for (int e = 0; text[e] != '\0'; e++)
    {
        if (text[e] == '.' || text[e] == '!' || text[e] == '?')
        {
            count_sentences++;
        }
    }

//Coleman-Liau index
    float L = 0;
    float S = 0;;
    float index = 0;
    float x = count_letters;
    float y = count_words;
    float z = count_sentences;

    L = 100 * (x / y);
    S = 100 * (z / y);

    index = (0.0588 * L) - (0.296 * S) - 15.8;
    int grade = round(index);

//Grade claasification
    if (grade < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (grade >= 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %i\n", grade);
    }
    return 0;
}